mealie/db/fixes/fix_migration_data.py

```diff
@@ -1,18 +1,73 @@
 from uuid import uuid4
 
 from slugify import slugify
+from sqlalchemy import and_, update
 from sqlalchemy.orm import Session
 
 from mealie.core import root_logger
+from mealie.db.models._model_base import SqlAlchemyBase
 from mealie.db.models.group.group import Group
 from mealie.db.models.household.shopping_list import ShoppingList, ShoppingListMultiPurposeLabel
 from mealie.db.models.labels import MultiPurposeLabel
 from mealie.db.models.recipe.ingredient import IngredientFoodModel, IngredientUnitModel
 from mealie.db.models.recipe.recipe import RecipeModel
+from mealie.db.models.users.users import User
 
 logger = root_logger.get_logger("init_db")
 
 
+def fix_dangling_refs(session: Session):
+    REASSIGN_REF_TABLES = ["group_meal_plans", "recipes", "shopping_lists"]
+    DELETE_REF_TABLES = ["long_live_tokens", "password_reset_tokens", "recipe_comments", "recipe_timeline_events"]
+
+    groups = session.query(Group).all()
+    for group in groups:
+        # Find an arbitrary admin user in the group
+        default_user = session.query(User).filter(User.group_id == group.id, User.admin == True).first()  # noqa: E712 - required for SQLAlchemy comparison
+        if not default_user:
+            # If there is no admin user, just pick the first user
+            default_user = session.query(User).filter(User.group_id == group.id).first()
+
+            # If there are no users in the group, we can't do anything
+            if not default_user:
+                continue
+
+        valid_user_ids = {user.id for user in group.users}
+
+        for table_name in REASSIGN_REF_TABLES:
+            table = SqlAlchemyBase.metadata.tables[table_name]
+            update_stmt = (
+                update(table)
+                .where(
+                    and_(
+                        ~table.c.user_id.in_(valid_user_ids),
```

Comment:
just for readability i would prefer notin_

```suggestion
                        table.c.user_id.notin_(valid_user_ids),
```